src/test/rgw/amqp_mock.cc

```diff
@@ -120,6 +121,24 @@ amqp_socket_t* amqp_tcp_socket_new(amqp_connection_state_t state) {
   return state->socket;
 }
 
+amqp_socket_t* amqp_ssl_socket_new(amqp_connection_state_t state) {
+  state->socket = new amqp_socket_t;
```

Comment:
maybe add another boolean flag to "amqp_socket_t_" indicating whether this is an ssl or regular socket.
later on in the test we can validate that to make sure that the new functionality works ok